### DISCUSSION TITLE
How to display authenticated user informations

### DISCUSSION THREAD

**Discussion by bucari** [rappasoft/laravel-livewire-tables#951] [Q&A]
Hi, i'm using laravel-livewire-tables to display authentication-log data, but i can't figure out how to display users informations.

If i join the users table to the Log::query in builder function, then the Column::make('User', 'users.name') doesn't work.

Any help?

Thanks

**Comment by tonypartridge** [comment]
I am not 100% sure I follow, is the users information stored in the log table? If not, have you joined over the log table in the QueryBuilder?

**Reply by bucari** [reply]
I joined the users table over the log table but Column::make returns an error.
So I think i'm doing something wrong.

**Reply by tonypartridge** [reply]
Can you provide the builder join you have added?

**Reply by bucari** [reply]
```
use Rappasoft\LaravelAuthenticationLog\Models\AuthenticationLog as Log;

public function builder(): Builder
{   
    return Log::query()
        ->join('users', 'authentication_log.authenticatable_id', '=', 'users.id')
        ->select('users.name');
}

public function columns(): array
{
    return [
        Column::make('Users', 'users.name')
              ->searchable(),
        Column::make('IP Address', 'ip_address')
            ->searchable(),
        Column::make('Browser', 'user_agent')
            ->searchable()
            ->format(function($value) {
                $agent = tap(new Agent, fn($agent) => $agent->setUserAgent($value));
                return $agent->platform() . ' - ' . $agent->browser();
            }),
        Column::make('Location')
            ->searchable(function (Builder $query, $searchTerm) {
                $query->orWhere('location->city', 'like', '%'.$searchTerm.'%')
                    ->orWhere('location->state', 'like', '%'.$searchTerm.'%')
                    ->orWhere('location->state_name', 'like', '%'.$searchTerm.'%')
                    ->orWhere('location->postal_code', 'like', '%'.$searchTerm.'%');
            })
            ->format(fn ($value) => $value && $value['default'] === false ? $value['city'] . ', ' . $value['state'] : '-'),
        Column::make('Login At')
            ->sortable()
            ->searchable()
            ->format(fn($value) => $value ? date('d/m/Y G:i:s',strtotime($value)) : '-'),
        Column::make('Login Successful')
            ->sortable()
            ->format(fn($value) => $value === true ? 'Si' : 'No'),
        Column::make('Logout At')
            ->sortable()
            ->searchable()
            ->format(fn($value) => $value ? date('d/m/Y G:i:s',strtotime($value)) : '-'),
    ];
}
```
The error is 
Call to undefined relationship [users] on model [Rappasoft\LaravelAuthenticationLog\Models\AuthenticationLog].

**Reply by tonypartridge** [reply]
So you you will get such an error as you have joined it without foresight it's looking for the relationship on the users model back to the Authentication  Log Model. So you should add a relationship onto the Authentication Log Model:

https://laravel.com/docs/9.x/eloquent-relationships

```php
public function users()
{
    return $this->belongsTo(User::class, 'foreign_key', 'owner_key');
}
```

As an example refined to your use case.

**Reply by bucari** [reply]
the Authentication Log Model has an authenticatable morph relationship
```
public function authenticatable(): MorphTo
{
    return $this->morphTo();
}
```
so in the authentication_log table, for example, i have authenticatable_type field that contains 'App\Models\User' and authenticatable_id 3 
How can i write the builder function to use authenticatable relation?
